k2/csrc/intersect_dense_pruned.cu

```diff
@@ -570,10 +573,17 @@ class MultiGraphDenseIntersectPruned {
           float best_loglike = max_per_fsa_data[i],
                 dynamic_beam = dynamic_beams_data[i];
           int32_t active_states = arc_end_scores_row_splits1_data[i + 1] -
-                                  arc_end_scores_row_splits1_data[i];
+                                  arc_end_scores_row_splits1_data[i],
+                  final_t = b_fsas_row_splits1[i+1] - b_fsas_row_splits1[i];
```

Comment:
Is this really final_t.. wouldn't final_t be b_fsas_row_splits1[i+1] - b_fsas_row_splits1[i] - 1, if we intend final_t to be the index of the last frame (i.e. the frame on which final-arcs happen)?

Reply:
I did it the same during my experiments, similar to this line (https://github.com/k2-fsa/k2/blob/510cd726f91d5c6473a37f1714cc13e57b9c056b/k2/csrc/intersect_dense_pruned.cu#L410).

Reply:
Ah, yes. final_t should be  `b_fsas_row_splits1[i+1] - b_fsas_row_splits1[i] - 1` . But I think this should not be the reason that making the results worse.